core/src/main/java/org/mule/api/MessagingException.java

```diff
@@ -146,8 +147,25 @@ protected String generateMessage(Message message)
                 payload = NullPayload.getInstance();
             }
 
-            buf.append(CoreMessages.messageIsOfType(payload.getClass()).getMessage());
-            addInfo("Payload", StringUtils.abbreviate(payload.toString(), 1000));
+            if (DefaultMuleConfiguration.verboseExceptions)
+            {
+                // Temporary workaround until MULE-4256 is implemented.
```

Comment:
I don't think that MULE-4256 will be fixed in the near future. Can you just move the "consumable" related code from DefaultMuleMessage to ClassUtils?